src/main/java/org/interledger/core/InterledgerAddressBuilder.java

```diff
@@ -0,0 +1,150 @@
+package org.interledger.core;
+
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+/**
+ * A Builder for immutable instances of {@link InterledgerAddress}.
+ * <p>
+ * <em>{@code InterledgerAddressBuilder} is not thread-safe and generally should not be stored in a field or collection,
+ * but instead used immediately to create instances.</em>
+ */
+public class InterledgerAddressBuilder {
+
+    private String value;
+
+    /**
+     * No-args Constructor.
+     */
+    public InterledgerAddressBuilder() {
+        // do nothing...
+    }
+
+    /**
+     * Constructor to enable copy-functionality.
+     */
+    public InterledgerAddressBuilder(final InterledgerAddress interledgerAddress) {
+        this.value = Objects.requireNonNull(interledgerAddress, "InterledgerAddress must not be null!").getValue();
+    }
+
+    /**
+     * Builder method to actually construct an instance of {@link InterledgerAddress} from the data in this builder.
+     *
+     * @return
+     */
+    public InterledgerAddress build() {
+        return new InterledgerAddressImpl(this);
+    }
+
+    /**
+     * Assign a new value to this builder.
+     *
+     * @param value A {@link String} representing this builder's "value", which is the string version of an Interledger
+     *              Address.
+     */
+    public InterledgerAddressBuilder withValue(final String value) {
+        this.value = Objects.requireNonNull(value, "value must not be null!");
+        return this;
+    }
+
+    /**
+     * Static helper method to construct a new builder.
+     *
+     * @return An instance of {@link InterledgerAddressBuilder}.
+     */
+    public static InterledgerAddressBuilder builder() {
+        return new InterledgerAddressBuilder();
+    }
+
+    /**
+     * A private, immutable implementation of {@link InterledgerAddress}.  To construct an instance of this class, use
+     * an instance of {@link InterledgerAddressBuilder}.
+     */
+    private static final class InterledgerAddressImpl implements InterledgerAddress {
+
+        private static final String REGEX = "(?=^.{1,1023}$)^(g|private|example|peer|self|test[1-3])[.]([a-zA-Z0-9_~-]+[.])*([a-zA-Z0-9_~-]+)?$";
```

Comment:
Does anyone know why it was decided to only allow a certain number of schemes?  On the one hand, the RFC talks about InterledgerAddresses being used for routing, and not having any central authority (sort of like, anything can work in here, as long as it has these alphanumeric characters).  

Currently, this implementation doesn't allow a scheme that isn't in the Regex, but it's unclear to me what value the schemes add.

Perhaps this is a question for the mailing list?